src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Represents a parser to process commands entered into the chatbot and returns the corresponding Command.
+ */
+public class Parser {
+
+    /**
+     * Parses the message to be processed into a Command, then returns that Command.
+     *
+     * @param message The message to be parsed.
+     * @return The Command parsed from the given message.
+     * @throws DukeException If there is an invalid input message causing an error in parsing.
+     */
+    public static Command parse(String message) throws DukeException {
+        if (message.equals("bye")) {
```

Comment:
I think this method could look a lot cleaner with the switch / case syntax. You could break it up into smaller methods as well

Reply:
thank you for the feedback! didn't realise that switch-case syntax can also be used for strings as well so will make the necessary changes 👍 . i agree that this parse method could be broken up into multiple methods since the ```parse``` function implementation is very long